src/nitsm/tsmcontext.py

```diff
@@ -217,7 +217,16 @@ def set_site_data(self, data_id: str, data: _Sequence[_Any]) -> None:
                 site_numbers property.
         """
 
-        return self._context.SetSiteData(data_id, data)
+        if data is not None and len(data) > 0:
```

Comment:
Unfortunately, site data is considerably more complicated than global data because the way sites are managed by the TSM context. Two TSM contexts might have different subsets of the sites on the system (e.g. [0,1] and [2,3]). Internally, TSM will maintain separate site data for all sites ([0,1,2,3]). In this example, each context would call set_site_data with two items that represents data for half of the sites. A subsequent step might have a TSM context with all sites in which case get_site_data must return 4 items.

I recommend fixing the problem with global data for now and postpone addressing site data until we have a better solution, possibly involving changes to TSM.